test/extensions/extension-manager.js

```diff
@@ -0,0 +1,326 @@
+
+import { checkErrorRevert, web3GetBalance } from "../../helpers/test-helper";
+import { setupEtherRouter } from "../../helpers/upgradable-contracts";
+import { setupColonyNetwork, setupMetaColonyWithLockedCLNYToken, setupRandomColony } from "../../helpers/test-data-generator";
+import { ROOT_ROLE, ARBITRATION_ROLE, ARCHITECTURE_ROLE, FUNDING_ROLE, ADMINISTRATION_ROLE } from "../../helpers/constants";
+
+const { expect } = chai;
+chai.use(bnChai(web3.utils.BN));
+
+const ExtensionManager = artifacts.require("ExtensionManager");
+const TestExtension0 = artifacts.require("TestExtension0");
+const TestExtension1 = artifacts.require("TestExtension1");
+const TestExtension2 = artifacts.require("TestExtension2");
+const Resolver = artifacts.require("Resolver");
+const IMetaColony = artifacts.require("IMetaColony");
+
+contract("ExtensionManager", accounts => {
+  let extensionManager;
+  let colonyNetwork;
+  let metaColony;
+  let colony;
+
+  let resolver0;
+  let resolver1;
+  let resolver2;
+
+  const ROOT = accounts[0];
+  const ARCHITECT = accounts[1];
+  const USER = accounts[2];
+
+  const TEST_EXTENSION = soliditySha3("TestExtension");
+
+  async function setupResolver(versionId) {
+    const resolver = await Resolver.new();
+    if (versionId === 0) {
+      const testExtension0 = await TestExtension0.new();
+      await setupEtherRouter("TestExtension0", { TestExtension0: testExtension0.address }, resolver);
+    } else if (versionId === 1) {
+      const testExtension1 = await TestExtension1.new();
+      await setupEtherRouter("TestExtension1", { TestExtension1: testExtension1.address }, resolver);
+    } else if (versionId === 2) {
+      const testExtension2 = await TestExtension2.new();
+      await setupEtherRouter("TestExtension2", { TestExtension2: testExtension2.address }, resolver);
+    }
+    return resolver;
+  }
+
+  before(async () => {
+    colonyNetwork = await setupColonyNetwork();
+    ({ metaColony } = await setupMetaColonyWithLockedCLNYToken(colonyNetwork));
+
+    await colonyNetwork.initialiseReputationMining();
+    await colonyNetwork.startNextCycle();
+
+    extensionManager = await ExtensionManager.new(metaColony.address);
+
+    resolver0 = await setupResolver(0);
+    resolver1 = await setupResolver(1);
+    resolver2 = await setupResolver(2);
+    await metaColony.addExtension(extensionManager.address, TEST_EXTENSION, resolver1.address, [FUNDING_ROLE, ADMINISTRATION_ROLE]);
+    await metaColony.addExtension(extensionManager.address, TEST_EXTENSION, resolver2.address, [FUNDING_ROLE, ADMINISTRATION_ROLE]);
+  });
+
+  beforeEach(async () => {
+    ({ colony } = await setupRandomColony(colonyNetwork));
+    await colony.setRootRole(extensionManager.address, true);
+    await colony.setArchitectureRole(1, 0, ARCHITECT, 1, true);
+  });
+
+  describe("without the extension manager", () => {
+    it("allows a colony to install an extension manually (and without a resolver)", async () => {
```

Comment:
I guess we can't stop this...

Reply:
Nor should we, I like that users can install the same extension without needing the extension manager.